lib/rabbit_feed/event.rb

```diff
@@ -70,6 +73,14 @@ def new_from_version_1 metadata_and_payload, schema
 
     private
 
+    def sensitive_proof_payload
+      clean_payload = payload.dup
+      sensitive_fields.each do |field|
+        clean_payload[field] = "[REMOVED]" if clean_payload.key?(field)
+      end
+      clean_payload
+    end
+
```

Comment:
Not a blocker, just offering as a suggestion _("less is more", and all that...)_

```ruby
    def sensitive_proof_payload
      sensitive_fields.each_with_object(payload.dup) do |field, clean_payload|
        clean_payload[field] = "[REMOVED]" if clean_payload.key?(field)
      end
    end
```